File: backend-python/app/modules/system/dto/print_template_request.py
from pydantic import BaseModel, Field


class PrintTemplateSaveRequest(BaseModel):
    code: str = Field(min_length=1, max_length=64)
    name: str = Field(min_length=1, max_length=100)
    paper_type: str = Field(alias="paperType", min_length=1, max_length=32)
    template_json: str = Field(alias="templateJson", min_length=1)
    status: int = Field(ge=0, le=1)


class PrintTemplateCopyRequest(BaseModel):
    code: str = Field(min_length=1, max_length=64)
    name: str = Field(min_length=1, max_length=100)
    status: int = Field(ge=0, le=1)

File: backend-rust/src/modules/system/dto/print_template_request.rs
use serde::Deserialize;
use serde_json::Value;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PrintTemplateSaveRequest {
    pub code: String,
    pub name: String,
    pub paper_type: Option<String>,
    pub template_json: String,
    pub status: i32,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PrintTemplateCopyRequest {
    pub code: String,
    pub name: String,
    pub status: i32,
}
